datatypes/src/raster/raster_tile.rs

```diff
@@ -124,27 +217,247 @@ where
+    }
+
+    #[test]
+    fn tile_information_global_upper_left_idx_2_3() {
+        let ti = TileInformation::new(
+            GridIdx([-2, 3]),
+            ArrayShape2D::from([100, 1000]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_upper_left_idx(), GridIdx([-200, 3000]));
+    }
+
+    #[test]
+    fn tile_information_global_upper_right_idx() {
+        let ti = TileInformation::new(
+            GridIdx([0, 0]),
+            ArrayShape2D::from([100, 100]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_upper_right_idx(), GridIdx([0, 99]));
+    }
+
+    #[test]
+    fn tile_information_global_upper_right_idx_2_3() {
+        let ti = TileInformation::new(
+            GridIdx([-2, 3]),
+            ArrayShape2D::from([100, 1000]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_upper_right_idx(), GridIdx([-200, 3999]));
+    }
+
+    #[test]
+    fn tile_information_global_lower_right_idx() {
+        let ti = TileInformation::new(
+            GridIdx([0, 0]),
+            ArrayShape2D::from([100, 100]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_lower_right_idx(), GridIdx([99, 99]));
+    }
+
+    #[test]
+    fn tile_information_global_lower_right_idx_2_3() {
+        let ti = TileInformation::new(
+            GridIdx([-2, 3]),
+            ArrayShape2D::from([100, 1000]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_lower_right_idx(), GridIdx([-101, 3999]));
+    }
+
+    #[test]
+    fn tile_information_global_lower_left_idx() {
+        let ti = TileInformation::new(
+            GridIdx([0, 0]),
+            ArrayShape2D::from([100, 100]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_lower_left_idx(), GridIdx([99, 0]));
+    }
+
+    #[test]
+    fn tile_information_global_lower_left_idx_2_3() {
+        let ti = TileInformation::new(
+            GridIdx([-2, 3]),
+            ArrayShape2D::from([100, 1000]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.global_lower_left_idx(), GridIdx([-101, 3000]));
+    }
+
+    #[test]
+    fn tile_information_local_to_global_idx_0_0() {
+        let ti = TileInformation::new(
+            GridIdx([0, 0]),
+            ArrayShape2D::from([100, 100]),
+            GeoTransform::default(),
+        );
+        assert_eq!(ti.local_to_global_idx(GridIdx([25, 75])), GridIdx([25, 75]));
+    }
+
+    #[test]
+    fn tile_information_local_to_global_idx_2_3() {
+        let ti = TileInformation::new(
+            GridIdx([-2, 3]),
+            ArrayShape2D::from([100, 1000]),
+            GeoTransform::default(),
+        );
+        assert_eq!(
+            ti.local_to_global_idx(GridIdx([25, 75])),
+            GridIdx([-175, 3075])
+        );
+    }
+
+    #[test]
+    fn tile_information_spatial_bounds() {
```

Comment:
would be great to have a test with a non default geo transform

Reply:
done